internal/controllers/topology/cluster/reconcile_state_test.go

```diff
@@ -269,6 +276,583 @@ func TestReconcileShim(t *testing.T) {
 	})
 }
 
+func TestReconcile_callAfterControlPlaneInitialized(t *testing.T) {
+	catalog := runtimecatalog.New()
+	_ = runtimehooksv1.AddToCatalog(catalog)
+
+	afterControlPlaneInitializedGVH, err := catalog.GroupVersionHook(runtimehooksv1.AfterControlPlaneInitialized)
+	if err != nil {
+		panic(err)
+	}
+
+	successResponse := &runtimehooksv1.AfterControlPlaneInitializedResponse{
+
+		CommonResponse: runtimehooksv1.CommonResponse{
+			Status: runtimehooksv1.ResponseStatusSuccess,
+		},
+	}
+	failureResponse := &runtimehooksv1.AfterControlPlaneInitializedResponse{
+		CommonResponse: runtimehooksv1.CommonResponse{
+			Status: runtimehooksv1.ResponseStatusFailure,
+		},
+	}
+
+	tests := []struct {
+		name               string
+		cluster            *clusterv1.Cluster
+		hookResponse       *runtimehooksv1.AfterControlPlaneInitializedResponse
+		wantMarked         bool
+		wantHookToBeCalled bool
+		wantError          bool
+	}{
+		{
+			name: "hook should be marked if the cluster is about to be created",
```

Comment:
nit: generally I think that hook marked or unmarked is kind of too technical, so I would suggest talking about tracking the intent to call a hook and removing the intent to call a hook, which is more verbose, but also explicit and easier to understand for people approaching the code for the first time
But this can be also a follow up improvement, not blocking now

```suggestion
			name: "intent to call hook should tracked if the cluster is about to be created",
```